site/content/entry/accurate-audio-emulation.md

```diff
@@ -0,0 +1,199 @@
+The first commit on Teakra started in late January 2018, but design and research phase started before this.
+There is some documentation online about the audio chip, the TeakLite, which [wwylele](https://github.com/wwylele) referenced often throughout development.
+[GBATek](http://problemkaputt.de/gbatek.htm#dsixpertteakdsp) is loaded with valuable information about the processor, put together by the reverse engineering efforts of Martin Korth and many other contributors.
+
+Building an LLE audio emulator from scratch is an exciting, yet somewhat scary prospect.
+When working on HLE audio emulation, you really start to understand what the games are doing, but when writing an LLE audio emulator, you create a magical black box where binary data goes in, and binary data comes out.
+This is especially true for [wwylele](https://github.com/wwylele) because he had never worked with a DSP before this!
+It's daunting to know that you'll gain a deep understanding of the architecture and the instruction set of the audio chip, but you won't gain any detailed knowledge of how the audio code is functioning.
+
+{twitch clip of reading disassembled firmwares}
+
+## Wait, How Long Until This Starts Playing Sound?
+
+Several months after starting, the interpreter was coming along nicely, and it was time to start hooking things together.
+This marked another very exciting yet very scary point in time; its been about 5 months of work on audio emulation, and all of it in complete silence!
+To understand why, first one must understand that audio hardware is rather complicated, and its more than just a simple interpreter.
+Anything beyond simple audio output is typically generated through specialized hardware known as a [Digital Signal Processor](https://en.wikipedia.org/wiki/Digital_signal_processor), or DSP for short.
+This custom built hardware is very efficient at transforming and processing audio samples while using less power, enabling the game developers to produce high quality audio without severely impacting battery life.
+In order to start seeing results, [wwylele](https://github.com/wwylele) needed to emulate the ways that the 3DS communicates with the DSP chip, such as [Memory Mapped IO](https://en.wikipedia.org/wiki/Memory-mapped_I/O), [hardware interrupts](https://en.wikipedia.org/wiki/Interrupt), and [Direct Memory Access](https://en.wikipedia.org/wiki/Direct_memory_access).
+The first two are a piece of cake compared to DMA; there is a straightfoward way to test how the 3DS and TeakLite DSP respond to them and recreate this in Teakra, and after a month of work, [wwylele](https://github.com/wwylele) got them functional.
+But DMA poses a unique challenge, as DMA enables the DSP to do direct memory to memory transfers, independent of what the CPU is doing, making it really hard to figure out how it works!
```

Comment:
It's probably also worth mentioning that DMA is the least documented part in gbatek doc among the three components you mentioned.